atc/worker/gardenruntime/worker.go

```diff
@@ -713,3 +640,100 @@ func (worker *Worker) GardenClient() gclient.Client {
 func (worker *Worker) BaggageclaimClient() baggageclaim.Client {
 	return worker.bcClient
 }
+
+func (worker *Worker) acquireVolumeStreamingLock(
+	ctx context.Context,
+	resourceCacheID int,
+	destination string,
+) (lock.Lock, bool, error) {
+	// if caching of streamed volumes is not enabled then we should not globally
+	// lock the volume streaming since steps cannot share the streamed volumes as caches
+	if !atc.EnableCacheStreamedVolumes {
+		return lock.NoopLock{}, true, nil
+	}
+
+	logger := lagerctx.FromContext(ctx)
+	return worker.db.LockFactory.Acquire(logger, lock.NewVolumeStreamingLockID(resourceCacheID, destination))
+}
+
+func (worker *Worker) findOrStreamVolume(
+	ctx context.Context,
+	privileged bool,
+	teamID int,
+	container db.CreatingContainer,
+	artifact runtime.Artifact,
+	mountPath string,
+	delegate runtime.BuildStepDelegate,
+) (Volume, error) {
+	logger := lagerctx.FromContext(ctx)
+	var tick *time.Ticker
+
+	_, inputPath := path.Split(mountPath)
+	if inputPath == "" {
+		inputPath = "for image"
```

Comment:
Sorry I'm not very familiar with the worker code, could you explain a bit as to what this is used for just for my own curiousity? Thanks!

Reply:
I think this is just for display in UI.